clippy_lints/src/casts/cast_possible_truncation.rs

```diff
@@ -153,5 +155,19 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, cast_expr: &Expr<'_>,
         _ => return,
     };
 
-    span_lint(cx, CAST_POSSIBLE_TRUNCATION, expr.span, &msg);
+    let snippet = snippet(cx, expr.span, "..");
+    let name_of_cast_from = snippet.split(" as").next().unwrap_or("..");
+    let suggestion = format!("{cast_to}::try_from({name_of_cast_from})");
```

Comment:
Using `cast_to` in the suggestion will lose any type aliases. It would be better to take a snippet.